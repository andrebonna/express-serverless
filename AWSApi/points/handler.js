'use strict';

module.exports.create = (event, context, callback) => {
    const response = {
        statusCode: 200,
        body: JSON.stringify({
            message: event
        })
    };
    callback(null, response);
};
