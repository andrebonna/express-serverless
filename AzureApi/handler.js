'use strict';

/* eslint-disable no-param-reassign */

module.exports.hello = function (context) {
  context.log('JavaScript HTTP trigger function processed a request.');

  context.res = {
    // status: 200, /* Defaults to 200 */
    body: JSON.stringify(context),
  };

  context.done();
};
