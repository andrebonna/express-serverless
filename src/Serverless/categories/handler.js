import React from 'react';
import templateBuilder from '../commons/templateBuilder';
import constants from '../commons/constants';
import Categories from '../../View/categories/Categories';

export function get(event, context, callback) {
    templateBuilder({
        title: 'Categories', 
        metas: constants.metas,
        props: {
            children: <Categories />
        }
    }, (err, data) => {
        const response = {
            statusCode: 200,
            headers: {
                'Content-type': 'text/html'
            },
            body: data
        };
        callback(null, response);
    });
}
