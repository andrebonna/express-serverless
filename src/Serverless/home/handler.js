import React from 'react';
import AWS from 'aws-sdk';
import templateBuilder from '../commons/templateBuilder';
import constants from '../commons/constants';
import Home from '../../View/home/Home';

const dynamodb = new AWS.DynamoDB.DocumentClient({ region: 'us-east-1' });

function getImagesFiltered(queryStringParameters) {
    let params = {
        TableName: 'Photos',
        Select: 'ALL_ATTRIBUTES'
    };

    if (queryStringParameters && queryStringParameters.filter) {
        params = {
            ...params,
            ScanFilter: {
                Categories: {
                    ComparisonOperator: 'CONTAINS',
                    AttributeValueList: [queryStringParameters.filter]
                }
            }
        };
    }

    return dynamodb.scan(params).promise();
}

function getCategories(data) {
    return dynamodb.scan({
        TableName: 'Photos',
        Select: 'SPECIFIC_ATTRIBUTES',
        AttributesToGet: ['Categories']
    }).promise().then((categories)=>{
        const categoriesSet = new Set();
        categories.Items.map(({ Categories })=>{
            Categories.forEach(category => categoriesSet.add(category));
        });          
        return { 
            data, 
            categories: [...categoriesSet] 
        };
    });
}

export function get(event, context, callback) {

    const {
        queryStringParameters
    } = event;

    getImagesFiltered(queryStringParameters)
        .then(getCategories)
        .then(({data, categories}) => {
            templateBuilder({
                title: 'Andr√©a Buck Photos',
                metas: constants.metas,
                props: {
                    categories,
                    children: <Home images={data.Items} />
                }
            }, (err, data) => {
                if (err) {
                    callback(err);
                }
                const response = {
                    statusCode: 200,
                    headers: {
                        'Content-type': 'text/html'
                    },
                    body: data
                };
                callback(null, response);
            });
        }).catch(callback);
}
