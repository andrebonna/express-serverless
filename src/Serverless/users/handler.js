import React from 'react';
import ejs from 'ejs';
import ReactDOMServer from 'react-dom/server';
import Head from '../../View/Head';
import App from '../../View/App';

export function get(event, context, callback) {

    const head = React.createElement(Head, {
        title: 'Serverless Universal React',
        metas: event.queryStringParameters
    });
    const data = {
        cdn_path: process.env.CDN_PATH || '',
        head: ReactDOMServer.renderToString(head),
        content: ReactDOMServer.renderToString(React.createElement(App))
    };
    ejs.renderFile(`${__dirname}/../../View/template.ejs`, data, function(err, data) {
        const response = {
            statusCode: 200,
            headers: {
                'Content-type': 'text/html'
            },
            body: data
        };
        callback(null, response);
    });
}

export function create(event, context, callback) {
    const response = {
        statusCode: 200,
        body: event.body
    };
    callback(null, response);
}
