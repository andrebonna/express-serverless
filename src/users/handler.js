'use strict';

module.exports.create = (event, context, callback) => {
    const response = {
        statusCode: 200,
        body: event.body
    };
    callback(null, response);
};
