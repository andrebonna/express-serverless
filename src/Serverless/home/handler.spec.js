import { get } from './handler';

describe('Handler Lambdas', ()=>{
    it('Test Get Home Page', (done) => {
        get({}, null, (_, response)=>{
            expect(response.body).toMatchSnapshot();
            done();
        });
    });
});
