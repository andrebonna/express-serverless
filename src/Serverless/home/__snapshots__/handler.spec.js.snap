// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handler Lambdas Test Get Home Page 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
    <head data-reactroot=\\"\\"><title>Andréa Buck Photos</title><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"bundle.css\\"/><meta charSet=\\"UTF-8\\"/><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\"/><meta name=\\"keywords\\" content=\\"photos, nature, urban, photography, fotografia\\"/></head>
    <body>
        <div id=\\"content\\"><div data-reactroot=\\"\\"><nav class=\\"navbar navbar-default navbar-fixed-top\\"><div class=\\"container-fluid\\"><div class=\\"navbar-header\\"><a href=\\"\\" class=\\"navbar-brand\\">React-Bootstrap</a><button type=\\"button\\" class=\\"navbar-toggle collapsed\\"><span class=\\"sr-only\\">Toggle navigation</span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span></button></div><div class=\\"navbar-collapse collapse\\"><ul class=\\"nav navbar-nav navbar-right\\"><li class=\\"dropdown\\"><a id=\\"basic-nav-dropdown\\" role=\\"button\\" class=\\"dropdown-toggle\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" href=\\"#\\">Categories<!-- --> <span class=\\"caret\\"></span></a><ul role=\\"menu\\" class=\\"dropdown-menu\\" aria-labelledby=\\"basic-nav-dropdown\\"><li role=\\"presentation\\" class=\\"\\"><a href=\\"categories\\" role=\\"menuitem\\" tabIndex=\\"-1\\">All</a></li><li role=\\"presentation\\" class=\\"\\"><a href=\\"categories?filter=nature\\" role=\\"menuitem\\" tabIndex=\\"-1\\">nature</a></li><li role=\\"presentation\\" class=\\"\\"><a href=\\"categories?filter=urban\\" role=\\"menuitem\\" tabIndex=\\"-1\\">urban</a></li></ul></li><li role=\\"presentation\\" class=\\"\\"><a role=\\"link\\" href=\\"contact\\">Contact Us</a></li></ul></div></div></nav><section class=\\"main-content\\"><div></div></section><footer class=\\"footer\\"><p>© André Bonna</p></footer></div></div>
        <script type=\\"text/javascript\\">
            window.APP_PROPS = '{\\"categories\\":[{\\"name\\":\\"All\\",\\"href\\":\\"categories\\"},{\\"name\\":\\"nature\\",\\"href\\":\\"categories?filter=nature\\"},{\\"name\\":\\"urban\\",\\"href\\":\\"categories?filter=urban\\"}],\\"children\\":{\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"images\\":[{\\"URLs\\":[\\"https://s3.amazonaws.com/andreabuck-photos/mockImage.jpg-1024.png\\",\\"https://s3.amazonaws.com/andreabuck-photos/mockImage.jpg-640.png\\",\\"https://s3.amazonaws.com/andreabuck-photos/mockImage.jpg-320.png\\"],\\"Categories\\":[\\"nature\\",\\"urban\\"],\\"Title\\":\\"Teste malucão de título\\",\\"Description\\":\\"bisse le bissè\\",\\"Name\\":\\"mockImage.jpg\\"}]},\\"_owner\\":null,\\"_store\\":{}},\\"page\\":\\"Home\\"}';
        </script>
        <script src=\\"bundle.js\\"></script>
    </body>
</html>"
`;
